docs/src/release_notes/21.8.0.rst

```diff
@@ -0,0 +1,97 @@
+    * not need any aggregation transforms (e.g. ``GroupByKey``, ``CombinePerKey``, ``Distinct``, etc). ``Reshuffle`` & ``Flatten`` are also not supported.
+    * not process non-KlioMessages (``klio-job.yaml::job_config.allow_non_klio_messages``)
+    * be deployed manually (i.e. via ``klio job run [OPTIONS]``).
+
+Logs can be viewed via **kubectl** [#f1]_ commands.
+Klio now supports configuration fields to enable the use of **shumway** [#f2]_, a library for sending metrics to a **FFWD** [#f3]_ agent.
+
+
+Metrics
+-------
+
+Default metrics [#f4]_ have been added for the following transform classes, with the name of the metric in parens:
+
+* IO transforms:
+    * ``KlioReadFromText`` (kmsg-read)
+    * ``KlioReadFromBigQuery`` (kmsg-read)
+    * ``KlioReadFromAvro`` (kmsg-read)
+    * ``KlioWriteToText`` (kmsg-write)
+    * ``KlioWriteToBigQuery`` (kmsg-write)
+    * ``KlioWriteToAvro`` (kmsg-write)
+    * ``KlioReadFromPubSub`` (kmsg-read)
+    * ``KlioWriteToPubSub`` (kmsg-write)
+* Helper Transforms:
+    * ``KlioGcsCheckInputExists`` (kmsg-data-found-input, kmsg-data-not-found-input)
+    * ``KlioGcsCheckOutputExists`` (kmsg-data-found-output, kmsg-data-not-found-output)
+    * ``KlioFilterPing`` (kmsg-process-ping, kmsg-skip-ping)
+    * ``KlioFilterForce`` (kmsg-process-force, kmsg-skip-force)
+    * ``KlioWriteToEventOutput`` (kmsg-output)
+    * ``KlioDrop`` (kmsg-drop)
+    * ``KlioCheckRecipients`` (kmsg-drop-not-recipient)
+    * ``KlioDebugMessage`` (kmsg-debug)
+    * ``KlioTriggerUpstream`` (kmsg-trigger-upstream)
+* Decorators:
+    * ``@retry`` (kmsg-retry-attempt, kmsg-drop-retry-error)
+    * ``@timeout`` (kmsg-drop-timed-out)
+    * ``@handle_klio`` (counters: kmsg-received, kmsg-drop-error, kmsg-success; timer: kmsg-timer)
+    * ``@serialize_klio_message`` (same metrics as @handle_klio`)
+
+These metrics also show up in Stackdriver monitoring when configured to use Stackdriver and are available to select in Metrics Explorer when building a dashboard.
+
+A user-facing metrics interface for shumway [#f2]_ has been added for when log-based metrics is not feasible.
+
+Changes
+-------
+
+:ref:`klio-cli <cli-21.8.0>`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. include:: /reference/cli/changelog.rst
+    :start-after: start-21.8.0
+    :end-before: end-21.8.0
+
+:ref:`klio <lib-21.8.0>`
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. include:: /reference/lib/changelog.rst
+    :start-after: start-21.8.0
+    :end-before: end-21.8.0
+
+
+:ref:`klio-core <core-21.8.0>`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. include:: /reference/core/changelog.rst
+    :start-after: start-21.8.0
+    :end-before: end-21.8.0
+
+:ref:`klio-exec <exec-21.8.0>`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. include:: /reference/executor/changelog.rst
+    :start-after: start-21.8.0
+    :end-before: end-21.8.0
+
+.. rubric:: Footnotes
+
+.. [#f0] https://docs.klio.io/en/stable/faqs/run_on_kubernetes.html
```

Comment:
This footnote link won't work on the build. What you should do is actually a doc reference within the text above:

```rst
Klio can now be run on the :doc:`additional runner of DirectGKERunner <faqs/run_on_kubernetes>`.
```

I think that works. If not, you can create a reference in the new FAQ page,

```rst
# in faqs/run_on_kubernetes.rst

.. _run-on-k8s:

How do I run a Klio job on Kubernetes?
=============================
```

Then in here:

```rst
Klio can now be run on the :ref:`additional runner of DirectGKERunner <run-on-k8s>`.
```

Reply:
Had to create a reference to get it to pass but sweet! the build passed and i checked that the link works!